src/core.js

```diff
@@ -241,63 +204,85 @@ define([
         },
     };
 
+    fn.getQuestionTypeGroupClass = function (slug) {
+        return "fd-question-group-" + slug;
+    };
+
     fn.getMugTypes = function () {
         return mugs.baseMugTypes;
     };
-        
-    fn._init_toolbar = function () {
+
+    fn._init_add_question = function() {
         var _this = this,
-            $questionGroupContainer = this.$f.find(
-                '.fd-container-question-type-group');
+            $addQuestion = this.$f.find(".fd-add-question");
 
         this.data.core.QUESTIONS_IN_TOOLBAR = [];
-        this.data.core.QUESTION_TYPE_TO_GROUP = {};
 
-        _.each(this._getQuestionGroups(), function (groupData) {
+        _.each(_this._getQuestionGroups(), function (groupData) {
             var groupSlug = groupData.group[0];
 
             var getQuestionData = function (questionType) {
                 var mugType = _this.data.core.mugTypes[questionType],
                     questionData = [
-                        questionType, 
-                        mugType.typeName, 
+                        questionType,
+                        mugType.typeName,
                         mugType.icon
                     ];
 
                 _this.data.core.QUESTIONS_IN_TOOLBAR.push(questionType);
-                _this.data.core.QUESTION_TYPE_TO_GROUP[questionType] = groupSlug;
                 return questionData;
             };
 
             groupData.questions = _.map(groupData.questions, getQuestionData);
-            if (groupData.related && groupData.related.length) {
-                groupData.related = _.map(groupData.related, getQuestionData);
-            }
+        });
 
-            groupData.group[2] = groupData.group[2] || 
-                _this.data.core.mugTypes[groupData.group[0]].icon;
-            $questionGroupContainer.append(
-                new QuestionTypeGroup(groupData, _this));
+        _this.$f.find(".fd-add-question").after($(add_question({
```

Comment:
minor: looks like there's a variable above called `$addQuestion` that already has this node.